io/net/bonding.py

```diff
@@ -0,0 +1,293 @@
+        if arg1 == "local":
+            self.log.info("Bonding configuration removed on laocal")
+            self.log.info("------------------------------------------------")
+            for ifs in self.host_interfaces:
+                cmd = "ifconfig %s down" % ifs
+                if process.system(cmd, shell=True, ignore_status=True) != 0:
+                    self.log.info("unable to bring down the interface")
+                cmd = "ifconfig %s down" % ifs
+                if process.system(cmd, shell=True, ignore_status=True) != 0:
+                    self.log.info("unable to bring down the interface")
+                cmd = "echo -%s > %s" % (ifs, self.bonding_slave_file)
+                if process.system(cmd, shell=True, ignore_status=True) != 0:
+                    self.log.info("bond removing failed in local machine")
+            cmd = "echo -%s > /sys/class/net/bonding_masters" % self.bond_name
+            if process.system(cmd, shell=True, ignore_status=True) != 0:
+                self.log.info("bond removing command failed in local machine")
+            time.sleep(5)
+        else:
+            self.log.info("Bonding configuration removed on Peer machine")
+            self.log.info("------------------------------------------------")
+            cmd = ''
+            cmd += 'ifconfig %s down;' % self.bond_name
+            for val in self.peer_interfaces:
+                cmd += 'ifconfig %s down;' % val
+            for val in self.peer_interfaces:
+                cmd += 'ip addr flush dev %s;' % val
+            for val in self.peer_interfaces:
+                cmd += 'echo "-%s" > %s;' % (val, self.bonding_slave_file)
+            cmd += 'echo "-%s" > /sys/class/net/bonding_masters;'\
+                   % self.bond_name
+            cmd += 'ifconfig %s %s netmask %s up;sleep 5;'\
+                   % (self.peer_if1, self.peer_ip1, self.net_mask[0])
+            peer_cmd = "ssh %s@%s \"%s\"" % (self.user, self.peer_ip1, cmd)
+            if process.system(peer_cmd, shell=True, ignore_status=True) != 0:
+                self.log.info("bond removing command failed in peer machine")
+
+    def ping_check(self, arg1):
+        '''
+        ping check
+        '''
+        cmd = "ping -I %s %s -c 5" % (self.bond_name, self.peer_ip1)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.fail("ping failed in Mode %s, check bonding configuration"
+                      % arg1)
+
+    def bond_fail(self, arg1):
+        '''
+        bond fail
+        '''
+        for interface in self.host_interfaces:
+            self.log.info("Failing interface %s for mode %s"
+                          % (interface, arg1))
+            cmd = "ifconfig %s down" % interface
+            if process.system(cmd, shell=True, ignore_status=True) != 0:
+                self.fail("bonding not working when trying to down the\
+                          interface %s " % interface)
+            time.sleep(15)
```

Comment:
that not good  as sleep should not be used @narasimhan-v can you please help us here